ironmon_tracker/Battle.lua

```diff
@@ -451,6 +525,14 @@ end
 function Battle.endCurrentBattle()
 	if not Battle.inBattle then return end
 
+	--Most of the time, Run Away message is present only after the battle ends
+	Battle.battleMsg = Memory.readdword(GameSettings.gBattlescriptCurrInstr)
+	if Battle.battleMsg == GameSettings.BattleScript_RanAwayUsingMonAbility then
+		local battleMon = Battle.BattleAbilities[0][Battle.Combatants[Battle.IndexMap[0]]]
+		local abilityOwner = Tracker.getPokemon(battleMon.abilityOwner.slot,battleMon.abilityOwner.isOwn)
+		Tracker.TrackAbility(abilityOwner.pokemonID, battleMon.ability)
+	end
```

Comment:
For what purpose is this check required? Prior to this PR, a player's own pokemon's ability would be track automatically as it is used in battle. So when a battle begins and their pokemon is sent out, if it has any ability at all, that ability is automatically tracked.

This check seems unnecessary, but I'm not sure if other code changes require it.

Reply:
This check is for tracking the ability Run Away. While a prompt appears on the screen for the ability, the battle ends as far as the tracker is concerned a few frames after the message appears, so most of the time it is not caught. So the check for the message needs to occur in the end battle script itself.
This for the case where you skill swap/transform into and enemy mon and then Run. If it was your own ability it would have already been tracked by the logic you mention, which should still be in place in the main battle update function.